packages/cli/src/utils/enableAutomine.ts

```diff
@@ -0,0 +1,107 @@
+import { debug } from "../debug";
+
+type MiningMode =
+  | {
+      type: "automine";
+    }
+  | {
+      type: "interval";
+      blockTime: number;
+    };
+
+export type EnableAutomineResult = { reset: () => Promise<void> };
+
+export async function enableAutomine(rpcUrl: string): Promise<EnableAutomineResult> {
+  try {
+    debug("Enabling automine");
+    const prevMiningMode = await getMiningMode(rpcUrl);
+    await setMiningMode(rpcUrl, { type: "automine" });
+    return {
+      reset: () => {
+        debug("Disabling automine");
+        return setMiningMode(rpcUrl, prevMiningMode);
+      },
+    };
+  } catch (e) {
+    debug("Skipping automine");
+    return { reset: async () => void 0 };
+  }
+}
+
+async function getMiningMode(rpcUrl: string): Promise<MiningMode> {
+  const { result: isAutomine } = await fetch(rpcUrl, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      jsonrpc: "2.0",
+      method: "anvil_getAutomine",
+      params: [],
+      id: 1,
+    }),
+  }).then((res) => res.json());
```

Comment:
why do we need to call the rpc manually when viem has actions for this? https://viem.sh/docs/actions/test/getAutomine

```ts
await getAction(client, getAutomine, 'getAutomine')({});
```

Reply:
agreed, much better now